games.py

```diff
@@ -255,7 +258,9 @@ def actions(self, state):
 
     def result(self, state, move):
         if move not in state.moves:
-            return state  # Illegal move has no effect
+            return GameState(to_move=('O' if state.to_move == 'X' else 'X'),
```

Comment:
Yes, I agree completely. I wondered the same thing. There were two reasons I chose such an implementation:

1. this `result` function is specific to the `TicTacToe` class, which inherits from `Game` - and thus won't apply to other `Game` classes. The `TicTacToe` class specifically mentions that the only valid states are 'X' and 'O' [here](https://github.com/aimacode/aima-python/blob/master/games.py#L245).
2. the same approach had already been used in the function, in line [268](https://github.com/aimacode/aima-python/blob/master/games.py#L268) below.

However, I would be happy to discuss alternative implementations. Maybe we could have a `turn` function defined in the `Game` class itself, which could then be appropriately overridden in all subclasses (including `TicTacTow`), thus effectively generalising this part of the code.